### DISCUSSION TITLE
Support excluding files/directories in the method `Commit.diff`

### DISCUSSION THREAD

**Discussion by dclong** [gitpython-developers/GitPython#1260] [Q&A] [answered]
The method `Commit.diff` supports a `paths` parameter to limit diff to specified paths. However, it is more convenient sometimes to exclude files/directories. And since this feature is supported by the `git` command, it makes to support excluding files/directories in the method `Commit.diff`.

**Comment by Byron** [comment, marked as answer]
Did you try to use `**kwargs` which will be passed to the `git diff` command directly? Invocations would look like `repo.head.commit.diff(diff_filter='a')`.

**Reply by hofbi** [reply]
If I want to explicitly exclude one file/folder from the git diff, I can get this via the CLI using `git diff <SHA> ':(exclude)path/to/exclude'` as suggested in https://stackoverflow.com/a/39943727.

How can I achieve this with `git.diff` since the exclude in not a named parameter. Hence I don't know what to specify for the `**kwargs`

**Reply by Byron** [reply]
Currently there is no support for using `**kwargs` for unnamed git flags, unfortunately excluding the suggested use case.
It's certainly possible to allow settings unnamed git flags through `**kwargs` with a set of special names, maybe like `git_arg_01="value"`.

**Reply by hofbi** [reply]
Thanks for your answer. It shomehow pointed me into the right direction. It just works via `*args`. So `git.diff(<SHA>, ":(exclude)path/to/exclude")` did it.
